icu4c/source/i18n/messageformat2_parser.cpp

```diff
@@ -209,16 +218,31 @@ UnicodeSet* initNameStartChars(UErrorCode& status) {
 
 UnicodeSet* initNameChars(UErrorCode& status) {
     if (U_FAILURE(status)) {
-        return {};
+        return nullptr;
     }
 
+    unisets::gUnicodeSets[unisets::NAME_START] = initNameStartChars(status);
```

Comment:
I think you can simplified to
```
UnicodeSet* nameStart = unisets::gUnicodeSets[unisets::NAME_START] = initNameStartChars(status);
UnicodeSet* digit = unisets::gUnicodeSets[unisets:: DIGIT] = initDigits(status);
if (U_FAILURE(status)) {
   return nullptr;
}
```

Reply:
Fixed in [e2990d2](https://github.com/unicode-org/icu/pull/3306/commits/e2990d2b9ccf2f31832348fa29d0f36ff4c3002f)